server/src/sim/financialanalysis.py

```diff
@@ -105,39 +105,48 @@ def financialanalysis(D, postyear=2015, S=None, makeplot=False, artgrowthrate=.0
     def accumu(lis):
         total = 0
         for x in lis:
-            total += x*D.opt.dt
+            total += x
             yield total
 
     # Store cost plot data
     for plottype in ['annual','cumulative']:
         plotdata[plottype] = {}
-        for plotsubtype in ['existing','total','future']:
-            plotdata[plottype][plotsubtype] = {}
-            if plottype=='annual':
-                for yscalefactor in costdisplays:
-                    plotdata[plottype][plotsubtype][yscalefactor] = {}
-                    plotdata[plottype][plotsubtype][yscalefactor]['xlinedata'] = simtvec
-                    plotdata[plottype][plotsubtype][yscalefactor]['xlabel'] = 'Year'
-                    plotdata[plottype][plotsubtype][yscalefactor]['title'] = 'Annual HIV-related costs - ' + plotsubtype + ' infections'
-                    if yscalefactor=='total':
-                        if not plotsubtype=='future': plotdata[plottype][plotsubtype][yscalefactor]['ylinedata'] = [(hivcosts[plotsubtype][j] + artcosts[plotsubtype][j])*(cpi[cpibaseyearindex]/cpi[j]) for j in range(noptpts)]
-                        plotdata[plottype][plotsubtype][yscalefactor]['ylabel'] = 'USD'
-                    else:
-                        if isinstance(sanitize(D.data.econ[yscalefactor].past[0]),int): continue #raise Exception('No data have been provided for this varaible, so we cannot display the costs as a proportion of this')
-                        yscale = expanddata(data=D.data.econ[yscalefactor].past[0], length=len(D.S.tvec)*D.opt.dt, growthrate=D.data.econ[yscalefactor].future[0][0], interp=True, dt=D.opt.dt)
-                        if not plotsubtype=='future': plotdata[plottype][plotsubtype][yscalefactor]['ylinedata'] = [(hivcosts[plotsubtype][j] + artcosts[plotsubtype][j])/yscale[j] for j in range(noptpts)] 
-                        plotdata[plottype][plotsubtype][yscalefactor]['ylabel'] = 'Proportion of ' + yscalefactor
+
+    plottype='annual'
```

Comment:
This will not work (at least not in normal Python), you still need to initialize e.g.  `plotdata['annual']` as well.
Tried it in Python console:
```
>>> plotdata = {}
>>> plottype='annual'
>>> for plotsubtype in ['existing','total','future']:
...     plotdata[plottype][plotsubtype]={}
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
KeyError: 'annual'
```